lib/pbench/server/templates.py

```diff
@@ -348,14 +360,58 @@ def add_mapping(self, name: str, body: Dict[AnyStr, Any]):
         """
         self.mappings.add_mapping(name, body)
 
-    def resolve(self):
+    def update(self, template: Template):
         """
-        In the initial implementation, this just loads the template files.
+        Update the template object from the database model object.
 
-        Placeholder: this will check the template DB and decide whether we
-        need to load the JSON files from disk based on modification dates.
+        Args:
+            template: DB Template instance
+        """
+        self.modified = template.mtime
+        self.version = template.version
+        self.mappings = JsonFile.raw_json(template.mappings)
+        self.settings = JsonFile.raw_json(template.settings)
+        self.template_name = template.template_name
+        self.index_pattern = template.index_template
+
+    def resolve(self):
         """
+        Check the on-disk template mapping files' modification dates against
+        the templates DB. Only if the template's base index name doesn't exist
+        in the DB, or if the DB object is older than the on-disk template do
+        we load the mapping files from disk.
+        """
+        try:
+            template = Template.find(self.name)
+        except TemplateNotFound:
+            template = None
+
+        # If we found a match that's newer, use it
+        if template and template.mtime >= self.modified:
+            self.update(template)
+            return
+
+        # If the DB version is missing, or older than the on-disk JSON, we need
+        # to fully resolve the mapping files. If we found an older DB object,
+        # update it with the new data, otherwise we create a new Template object.
         self.load()
+        if not template:
+            template = Template(
+                name=self.name,
+                template_name=self.template_name,
+                file=str(self.mappings.file),
+                index_template=self.index_pattern,
+                settings=self.settings.json,
+                mappings=self.mappings.json,
+                version=self.version,
+            )
```

Comment:
You're not specifying the modification time.  I think this creates an opportunity for a pratfall:  if we do a run with an old version of a template, and then we go "oops" and try to do a run with the correct version of the template, the database won't get updated unless the correct version has a newer date than _when we did the initial run_.  So, since we have the template file here, I think we should use _its_ date.